warehouse/integrations/datalake/datalake_test.go

```diff
@@ -407,4 +417,145 @@ func TestIntegration(t *testing.T) {
 		}
 		testhelper.VerifyConfigurationTest(t, dest)
 	})
+
+	t.Run("Trino", func(t *testing.T) {
+		dsn := fmt.Sprintf("http://user@localhost:%d?catalog=minio&schema=default&session_properties=minio.parquet_use_column_index=true",
+			trinoPort,
+		)
+		db, err := sql.Open("trino", dsn)
+		require.NoError(t, err)
+
+		require.Eventually(t, func() bool {
+			_, err := db.ExecContext(ctx, "SELECT 1")
+			return err == nil
+		}, 60*time.Second, 1*time.Second)
+
+		require.NoError(t, testhelper.WithConstantRetries(func() error {
+			_, err = db.ExecContext(ctx, `
+				CREATE SCHEMA IF NOT EXISTS minio.rudderstack WITH (
+				location = 's3a://`+s3BucketName+`/')
+			`)
+			return err
+		}))
+
+		require.NoError(t, testhelper.WithConstantRetries(func() error {
+			_, err = db.ExecContext(ctx, `
+				CREATE TABLE IF NOT EXISTS minio.rudderstack.tracks (
+					"_timestamp" TIMESTAMP,
+					context_destination_id VARCHAR,
+					context_destination_type VARCHAR,
+					context_ip VARCHAR,
+					context_library_name VARCHAR,
+					context_passed_ip VARCHAR,
+					context_request_ip VARCHAR,
+					context_source_id VARCHAR,
+					context_source_type VARCHAR,
+					event VARCHAR,
+					event_text VARCHAR,
+					id VARCHAR,
+					original_timestamp TIMESTAMP,
+					received_at TIMESTAMP,
+					sent_at TIMESTAMP,
+					"timestamp" TIMESTAMP,
+					user_id VARCHAR,
+					uuid_ts TIMESTAMP
+				)
+				WITH (
+					external_location = 's3a://`+s3BucketName+`/some-prefix/rudder-datalake/s_3_datalake_integration/tracks/2023/05/12/04/',
+					format = 'PARQUET'
+				)
+			`)
+			return err
+		}))
+
+		var count int64
+
+		require.NoError(t, testhelper.WithConstantRetries(func() error {
```

Comment:
I suggest using `require.Eventually` with a short interval here as well, for sure one that is a fraction of a second. Same applies for others below.